### PR TITLE
Improve SignalR idle timeout example

### LINKED ISSUE #32021: IdleCircuitHandler should log a meaningful log message
### Description

Example for [Monitor server-side circuit activity
](https://learn.microsoft.com/en-us/aspnet/core/blazor/fundamentals/signalr?view=aspnetcore-8.0#monitor-server-side-circuit-activity) contains below code, which prints a log message, when circuit is detected to be idle:

```
private void CircuitIdle(object? sender, System.Timers.ElapsedEventArgs e)
{
        logger.LogInformation("{Circuit} is idle", nameof(CircuitIdle));
}
```

Above code, when triggered, would print `CircuitIdle is dile`, which is not a very helpful log message. A more meaningful message would include the ID of the circuit that has become idle.

### Page URL

https://learn.microsoft.com/en-us/aspnet/core/blazor/fundamentals/signalr?view=aspnetcore-8.0

### Content source URL

https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/blazor/fundamentals/signalr.md

### Document ID

d17e4c28-ea82-0dd8-295d-19f6ada147ea

### Article author

@guardrex

### PR DESCRIPTION
**Pull request by guardrex**
Fixes  #32021

Thanks @gerektoolhy! 🚀 ... This will log the ID ***and*** fix a few other NITs.

@MackinnonBuck ... The updates ***seem*** reasonable 🤔, but I'll wait for you to look because I don't think you saw the original content when it went in. This will give you a chance to look over the entire section in case there are additional remarks to make (e.g., ***gotchas*** 😈).

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [aspnetcore/blazor/fundamentals/signalr.md](https://github.com/dotnet/AspNetCore.Docs/blob/237fc677446c2dd06321e9deec0cfb61c284dc28/aspnetcore/blazor/fundamentals/signalr.md) | [ASP.NET Core Blazor SignalR guidance](https://review.learn.microsoft.com/en-us/aspnet/core/blazor/fundamentals/signalr?branch=pr-en-us-32027) |

### CODE REVIEW

File: aspnetcore/blazor/fundamentals/signalr.md [hunk 1]
```diff
@@ -650,27 +650,43 @@ By default, components are prerendered on the server before the client connectio
 
 Monitor inbound circuit activity using the <xref:Microsoft.AspNetCore.Components.Server.Circuits.CircuitHandler.CreateInboundActivityHandler%2A> method on <xref:Microsoft.AspNetCore.Components.Server.Circuits.CircuitHandler>. Inbound circuit activity is any activity sent from the browser to the server, such as UI events or JavaScript-to-.NET interop calls.
 
-For example, you can use a circuit activity handler to detect if the client is idle:
+For example, you can use a circuit activity handler to detect if the client is idle and log its circuit ID (<xref:Microsoft.AspNetCore.Components.Server.Circuits.Circuit.Id?displayProperty=nameWithType>):
 
 ```csharp
+using Microsoft.AspNetCore.Components.Server.Circuits;
+using Microsoft.Extensions.Options;
+using Timer = System.Timers.Timer;
```
Comment (MackinnonBuck) [inline aspnetcore/blazor/fundamentals/signalr.md:658]: Is `using System.Timers;` insufficient? Or does that create ambiguity in the code somewhere?

File: aspnetcore/blazor/fundamentals/signalr.md [hunk 2]
```diff
@@ -650,27 +650,43 @@ By default, components are prerendered on the server before the client connectio
...
-        logger.LogInformation("{Circuit} is idle", nameof(CircuitIdle));
+        logger.LogInformation("{CircuitId} is idle", currentCircuit?.Id);
+    }
+
+    public override Task OnCircuitOpenedAsync(Circuit circuit, 
+        CancellationToken cancellationToken)
+    {
+        currentCircuit = circuit;
+
+        return base.OnCircuitOpenedAsync(circuit, cancellationToken);
```
Comment (MackinnonBuck) [inline aspnetcore/blazor/fundamentals/signalr.md:689]: It's not necessary to call the base method here - it doesn't have a meaningful implementation and just returns `Task.CompletedTask`.

**Review by MackinnonBuck — APPROVED** [review]
Looks good! Just a few questions, feel free to address however you like

File: aspnetcore/blazor/fundamentals/signalr.md (hunk 1 above)
Comment (guardrex) [reply aspnetcore/blazor/fundamentals/signalr.md:658]: I took that from Dan's code. It looks like going with `using System.Timers.Timer;` is ambiguous with `System.Threading.Timer`. I suppose that's why he did it this way.

Here's Dan's code btw ---

https://devblogs.microsoft.com/dotnet/asp-net-core-updates-in-dotnet-8-preview-3/#monitor-blazor-server-circuit-activity

File: aspnetcore/blazor/fundamentals/signalr.md (hunk 2 above)
Comment (guardrex) [reply aspnetcore/blazor/fundamentals/signalr.md]: If I don't return, I get an error that the `OnCircuitOpenedAsync` isn't returning a value. How is it set up without returning the base method?

Comment (guardrex) [reply aspnetcore/blazor/fundamentals/signalr.md]: Make it just return a completed `Task`?

Comment (MackinnonBuck) [reply aspnetcore/blazor/fundamentals/signalr.md]: Yep, returning `Task.CompletedTask` is the right move!

File: aspnetcore/blazor/fundamentals/signalr.md (hunk 1 above)
Comment (MackinnonBuck) [reply aspnetcore/blazor/fundamentals/signalr.md:658]: In that case, it's fine how it is!

**Pull request merged**